### PR TITLE
Proofread parallel manual/docs

### PR DESCRIPTION
**Pull request by gpeairs**
@kshyatt asked me to read the parallel computing documentation. These are mostly minor changes and clarifications for the things I stumbled on (as a reader with some background in parallel stuff but less in Julia).

My main remaining note is that the two sections "Remote References and [X]" could benefit from examples. Also, "Synchronization with Remote References" is an empty section, and I don't know if there was a plan with that.

I was sitting on the first two commits for a while and just rebased. @amitmurthy made some changes to the Channel constructor in the meantime, so it looks like I reverted the docstring changes in my second commit and put them back later. I'm new at this, so I may have messed something up.

### CODE REVIEW

File: doc/manual/parallel-computing.rst [hunk 1]
```diff
@@ -507,7 +509,7 @@ Every time a :class:`RemoteChannel` or a (unfetched) :class:`Future` is serializ
 to by the reference is notified. And every time a :class:`RemoteChannel` or a (unfetched) :class:`Future`
 is garbage collected locally, the node owning the value is again notified.
 
-The notifications are done via sending of "tracking" messages - an "add reference" message when
+The notifications are done via sending of "tracking" messages---an "add reference" message when
```
Comment (amitmurthy) [inline doc/manual/parallel-computing.rst]: What does `---` do in rst?

File: doc/manual/parallel-computing.rst [hunk 2]
```diff
@@ -461,15 +463,15 @@ via :func:`fetch` and :func:`take!`. Multiple writers can add to the :class:`Cha
 the channel, while :func:`wait` waits for an object to become available.
 :func:`close` closes a :class:`Channel`. On a closed :class:`Channel`\, :func:`put!` will fail,
 while :func:`take!` and :func:`fetch` successfully return any existing values
-till it is emptied.
+until it is emptied.
```
Comment (amitmurthy) [inline doc/manual/parallel-computing.rst]: "till" and "until" mean the same, but yes, whatever is preferred.

File: doc/manual/parallel-computing.rst [hunk 3]
```diff
@@ -799,18 +801,18 @@ starting workers on the same host, is implemented::
...
 
 The :func:`launch` method is called asynchronously in a separate task. The termination of this task
 signals that all requested workers have been launched. Hence the :func:`launch` function MUST exit as soon
 as all the requested workers have been launched.
 
-Newly launched workers are connected to each other, and the master process, in a all-to-all manner.
+Newly launched workers are connected to each other, and the master process, in an all-to-all manner.
 Specifying the command argument ``--worker <cookie>`` results in the launched processes initializing themselves
-as workers and connections being setup via TCP/IP sockets. Optionally, ``--bind-to bind_addr[:port]``
+as workers and connections being set up via TCP/IP sockets. Optionally, ``--bind-to bind_addr[:port]``
```
Comment (amitmurthy) [inline doc/manual/parallel-computing.rst]: Googling reports that `setup` is apparently the preferred form in technical literature.

**Comment by amitmurthy** [comment]
Content looks good to me.

File: doc/manual/parallel-computing.rst (hunk 3 above)
Comment (tkelman) [inline doc/manual/parallel-computing.rst]: Depends on usage, I think this referring to the state reads better as "being set up" than "being setup."

File: doc/manual/parallel-computing.rst [hunk 4]
```diff
@@ -880,22 +882,22 @@ times during the worker's lifetime with appropriate ``op`` values:
...
-- Each task handles all incoming messages from a single remote worker in a message processing loop
-- The message processing loop waits on an :class:`IO` object (for example, a :class:`TCPSocket` in the default implementation), reads an entire
-  message, processes it and waits for the next one
-- Sending messages to a process is done directly from any Julia task - not just communication tasks - again, via the appropriate
-  :class:`IO` object
+- Each Julia process thus has 31 communication tasks.
+- Each task handles all incoming messages from a single remote worker in a message-processing loop.
+- The message-processing loop waits on an :class:`IO` object (for example, a :class:`TCPSocket` in the default implementation), reads an entire
+  message, processes it and waits for the next one.
+- Sending messages to a process is done directly from any Julia task---not just communication tasks---again, via the appropriate
```
Comment (gpeairs) [inline doc/manual/parallel-computing.rst]: Yes, an em dash, but maybe using a Unicode em dash in the first place would be better.

**Comment by amitmurthy** [comment]
Thanks.

File: doc/manual/parallel-computing.rst (hunk 4 above)
Comment (jiahao) [inline doc/manual/parallel-computing.rst]: Putting Unicode in the docs wreaks havoc with the LaTeX/PDF generator, thanks to sphinx being hard-coded to use only pdflatex.

Comment (tkelman) [inline doc/manual/parallel-computing.rst]: good thing there's an open pr that replaces sphinx then

**Pull request merged**